src/store/actionCreators/SavedQueries.ts

```diff
@@ -0,0 +1,100 @@
+import { ThunkAction } from 'redux-thunk';
+
+import { deleteSavedQuery, fetchAllSavedQueries } from 'services/riffQueries';
+import { getVirtualStudies } from 'services/virtualStudies';
+
+import { Api } from '../apiTypes';
+import { RootState } from '../rootState';
+import {
+  QueryType,
+  SavedQueriesActions,
+  SavedQueriesActionTypes,
+  SplitSavedQueries,
+} from '../SavedQueriesTypes';
+import { LoggedInUser } from '../userTypes';
+
+import { deleteVirtualStudy } from './virtualStudies';
+
+const toggleFetchAllQueriesLoading = (isLoadingAllQueries: boolean): SavedQueriesActionTypes => ({
+  type: SavedQueriesActions.toggleFetchAllSavedQueries,
+  isLoadingAllQueries,
+});
+
+const addQueries = (queries: SplitSavedQueries): SavedQueriesActionTypes => ({
+  type: SavedQueriesActions.addAllSavedQueries,
+  queries,
+});
+
+const failureFetchingAllSavedQueries = (error: Error | null): SavedQueriesActionTypes => ({
+  type: SavedQueriesActions.failureFetchingAllSavedQueries,
+  error,
+});
+
+const requestDeleteSavedQueryAction = (queryId: string): SavedQueriesActionTypes => ({
+  type: SavedQueriesActions.requestDeleteSavedQuery,
+  queryId,
+});
+
+const successDeletingSavedQueryAction = (
+  queryId: string,
+  queryType: QueryType,
+): SavedQueriesActionTypes => ({
+  type: SavedQueriesActions.successDeletingSavedQuery,
+  queryId,
+  queryType,
+});
+
+const failureDeletingQuery = (queryId: string, error: Error | null): SavedQueriesActionTypes => ({
```

Comment:
I don't see a case where error would be null

Reply:
Good point, I'll change that.